### ISSUE TITLE
Deprecated Warning using Invoker call method

### ISSUE THREAD

**Issue by levidurfee** [PHP-DI/PHP-DI#613]
```php
// Csrf.php
class Csrf 
{
  public function run()
  {
    die('i made it here');
  }
}


// Caller.php
use Invoker\Invoker;

class Caller
{
  public function __construct(ContainerInterface $container, Invoker $invoker)
  {
    $this->container = $container;
    $this->invoker = $invoker;
  }
  public function call()
  {
    $this->invoker->call(['Levi\Csrf', 'run'])
  }
}

```

`Deprecated: call_user_func_array() expects parameter 1 to be a valid callback, non-static method Csrf::run() should not be called statically in /Users/levidurfee/projects/.../vendor/php-di/invoker/src/Invoker.php on line 82`

I redacted some elements of the classes, but I hope this provides enough context to help you understand what is happening. If it doesn't, then I'll be happy to create a smaller project to demonstrate the issue, one that I can share publicly.

If no one has time to look into this issue, then I'll be more than happy to try and fix it.

**Comment by mnapoli** [comment]
Thank you for your report. You are using the `Invoker` class to invoke `run()` (https://github.com/PHP-DI/Invoker). By default `Invoker` does not instantiate classes to invoke non-static methods.

How is the `Invoker` configured? Do you want to use a custom `Invoker` instance (because the Container class also implements `InvokerInterface` and *does* instantiate classes).

With a bit more info I can help more :)

**Comment by levidurfee** [comment]
I've tried using the Invoker and the Container. I should have mentioned that this only occurs with PHP 7.2.x.

I configured the Invoker like this.

```php
$builder = new \DI\ContainerBuilder();
$builder->addDefinitions('../config.php');
$container = $builder->build();
$invoker = new Invoker\Invoker(null, $container);

$app = new Application($invoker, $container);
```

Before that, I passed the `$container` into my `Application` class twice, but type-hinted as two different types. I used the `Invoker\InvokerInterface` and `Psr\Container\ContainerInterface`.

```php
$builder = new \DI\ContainerBuilder();
$builder->addDefinitions('../config.php');
$container = $builder->build();

$app = new Application($container, $container);
```

I've created an [example repo](https://github.com/levidurfee/phpdi-example) that demonstrates the issue. The solution might be to refactor/restructure my code because it's already confusing me and I'm not sure there is any issue with PHP-DI. I appreciate your help :)
